frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx

```diff
@@ -0,0 +1,535 @@
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  describe("list values", () => {
+    const column = findColumn("ORDERS", "QUANTITY");
+
+    it("should allow to pick a list value", async () => {
+      const { onChange } = await setupNumberPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+        fieldValues: createMockFieldValues({
+          field_id: ORDERS.QUANTITY,
+          values: [[10], [20], [30]],
+        }),
+      });
+
+      userEvent.click(screen.getByText("20"));
+
+      expect(onChange).toHaveBeenCalledWith([20]);
+    });
+
+    it("should handle field values remapping", async () => {
+      const { onChange } = await setupNumberPicker({
+        query,
+        stageIndex,
+        column,
+        values: [10],
+        fieldValues: createMockFieldValues({
+          field_id: ORDERS.QUANTITY,
+          values: [
+            [10, "To-do"],
+            [20, "In-progress"],
+            [30, "Completed"],
+          ],
+        }),
+      });
+      expect(screen.getByRole("checkbox", { name: "To-do" })).toBeChecked();
+      expect(screen.getByRole("checkbox", { name: "To-do" })).toBeChecked();
+
+      userEvent.type(screen.getByPlaceholderText("Search the list"), "in");
+      expect(screen.getByText("In-progress")).toBeInTheDocument();
+      expect(screen.queryByText("Completed")).not.toBeInTheDocument();
+
+      userEvent.click(screen.getByText("In-progress"));
+      expect(onChange).toHaveBeenCalledWith([10, 20]);
+    });
+  });
+
+  describe("no values", () => {
+    const column = findColumn("PEOPLE", "PASSWORD");
+
+    it("should allow to add a value", async () => {
+      const { onChange, onFocus, onBlur } = await setupNumberPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+      });
+
+      const input = screen.getByPlaceholderText("Enter a number");
+      userEvent.type(input, "123");
+      userEvent.tab();
+
+      expect(onFocus).toHaveBeenCalled();
+      expect(onChange).toHaveBeenLastCalledWith([123]);
+      expect(onBlur).toHaveBeenCalled();
+    });
+
+    it("should not allow to add empty values", async () => {
+      const { onChange, onFocus, onBlur } = await setupNumberPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+      });
+
+      const input = screen.getByPlaceholderText("Enter a number");
+      userEvent.type(input, "123");
+      userEvent.clear(input);
+      userEvent.tab();
+
+      expect(onFocus).toHaveBeenCalled();
+      expect(onChange).toHaveBeenLastCalledWith([]);
+      expect(onBlur).toHaveBeenCalled();
+    });
+
+    it("should not allow to add invalid values", async () => {
+      const { onChange, onFocus, onBlur } = await setupNumberPicker({
+        query,
+        stageIndex,
+        column,
+        values: [],
+      });
+
+      const input = screen.getByPlaceholderText("Enter a number");
+      userEvent.type(input, "abc");
```

Comment:
I think it's worth asserting that whitespace-only string will be rejected as well.
I remember the initial implementation would treat `" "` as `0`.

Reply:
Fixed